### PR TITLE
Temporarily import `loop_in_thread` fixture from `distributed`

### PR DESCRIPTION
**Pull request by jrbourbeau**
- [x] Closes https://github.com/dask/distributed/issues/6842
- [ ] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

This is a quickfix similar to https://github.com/dask/dask/pull/9337. Note there's a conversation happening over in https://github.com/dask/distributed/issues/6806 around what testing utilities in `distributed` should be considered public.

### CODE REVIEW

File: xarray/tests/test_distributed.py [hunk 1]
```diff
@@ -14,7 +14,13 @@
 
 from dask.distributed import Client, Lock
 from distributed.client import futures_of
-from distributed.utils_test import cluster, gen_cluster, loop, cleanup  # noqa: F401
+from distributed.utils_test import (  # noqa: F401
+    cluster,
+    gen_cluster,
+    loop,
+    cleanup,
+    loop_in_thread,
```
Comment (jrbourbeau) [inline xarray/tests/test_distributed.py:22]: This is only available in newer versions of `distributed` -- what's the minimum version of `distributed` `xarray` supports?

Comment (andersy005) [reply xarray/tests/test_distributed.py:22]: i believe we are now supporting `distributed v2021.4+`: https://github.com/pydata/xarray/blob/main/ci/requirements/min-all-deps.yml

Comment (jrbourbeau) [reply xarray/tests/test_distributed.py:22]: Thanks @andersy005! Actually looking at it more closely, it looks like `loop_in_thread` has been around for a long time but the need to actually `import` it is new. So the changes here should both fix the failures we're seeing and also be backwards compatible. It also looks like there's a CI build which tests against the minimum supported version of packages (including `distributed`)

https://github.com/pydata/xarray/blob/787a96c15161c9025182291b672b3d3c5548a6c7/.github/workflows/ci.yaml#L52-L54

which is also passing

**Comment by jrbourbeau** [comment]
It's not immediately clear to me why the docs build is failing. Has that been happening elsewhere?

**Comment by andersy005** [comment]
> It's not immediately clear to me why the docs build is failing. Has that been happening elsewhere?

we turn sphinx warnings into errors when building docs on RTD. It appears the SSL certificate for numba.pydata.org has expired and sphinx is throwing a warning which is then turned into an error after docs build. 


```bash
loading intersphinx inventory from https://rasterio.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://sparse.pydata.org/en/latest/objects.inv...
WARNING: failed to reach any of the inventories with the following issues:
intersphinx inventory 'https://numba.pydata.org/numba-doc/latest/objects.inv' not fetchable due to <class 'requests.exceptions.SSLError'>: HTTPSConnectionPool(host='numba.pydata.org', port=443): Max retries exceeded with url: /numba-doc/latest/objects.inv (Caused by SSLError(CertificateError("hostname 'numba.pydata.org' doesn't match either of '*.github.com', 'www.github.com', 'github.io', 'github.com', '*.github.io', 'githubusercontent.com', '*.githubusercontent.com'")))
```

**Comment by andersy005** [comment]
I opened an issue just in case this is something that can be fixed upstream. 

- https://github.com/numba/numba.github.com/issues/10

**Comment by andersy005** [comment]
Thank you for this fix, @jrbourbeau!

**Comment by dcherian** [comment]
Thanks @jrbourbeau!
